README.md
# Cardinal Numerals

![Cardinal Numbers (Tungstenblue/Getty Images)](cardinal_numbers.png)

A [cardinal numeral](<https://en.wikipedia.org/wiki/Cardinal_number_(linguistics)>) is a noun used to count, such as the Vietnamese words **một**, **hai**, **ba**, but also compounds, such as **ba trăm bốn mươi hai**.

# Waypoint 1: Generate Vietnamese Numerals

Write a function `integer_to_vietnamese_numeral` that takes an argument `n` representing a positive integer, and returns a string corresponding to the **North** [Vietnamese cardinal numeral](https://en.wikipedia.org/wiki/Vietnamese_numerals) of this number.

The function raises the exception `TypeError` if the argument `n` is not an integer.

The function raises the exception `ValueError` if the argument `n` is not positive integer.

The maximum integer the function can generate a cardinal numeral is `999,999,999,999`. If `n` is over this number, the function raises the exception `OverflowError`.

For example:

```python
>>> integer_to_vietnamese_numeral(96)
'chín mươi sáu'
>>> integer_to_vietnamese_numeral(405)
'bốn trăm linh năm'
>>> integer_to_vietnamese_numeral(1915)
'một nghìn chín trăm mười lăm'
>>> integer_to_vietnamese_numeral(5061)
'năm nghìn không trăm sáu mươi mốt'
>>> integer_to_vietnamese_numeral(1002003)
'một triệu không trăm linh hai nghìn không trăm linh ba'
>>> integer_to_vietnamese_numeral(1000000)
'một triệu'
>>> integer_to_vietnamese_numeral(1030000)
'một triệu không trăm ba mươi nghìn'
>>> integer_to_vietnamese_numeral(1002003)
'một triệu không trăm linh hai nghìn không trăm linh ba'
>>> integer_to_vietnamese_numeral(1002003004)
'một tỷ không trăm linh hai triệu không trăm linh ba nghìn không trăm linh bốn'
>>> integer_to_vietnamese_numeral(1002003004)
'một tỷ không trăm linh ba nghìn không trăm linh bốn'
>>> integer_to_vietnamese_numeral(1002000004)
'một tỷ không trăm linh hai triệu không trăm linh bốn'
>>> integer_to_vietnamese_numeral(100000004)
'một trăm triệu không trăm linh bốn'
>>> integer_to_vietnamese_numeral(999999999999)
'chín trăm chín mươi chín tỷ chín trăm chín mươi chín triệu chín trăm chín mươi chín nghìn chín trăm chín mươi chín'
>>> integer_to_vietnamese_numberal('9999999999990')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "<input>", line 4, in _integer_to_vietnamese_numeral
Overflow: Integer greater than 999,999,999,999
>>> integer_to_vietnamese_numberal('4096')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "<input>", line 3, in _integer_to_vietnamese_numeral
TypeError: Not an integer
>>> integer_to_vietnamese_numeral(-1)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "<input>", line 6, in _integer_to_vietnamese_numeral
ValueError: Not a positive integer
```

_Hint: if you have not a clear idea how to start, you should consider revising the [international system of numeration](https://www.youtube.com/watch?v=jKo_NrYcAaA), which is followed by most of the countries of the world, including Vietnam. In this system, a number is split up into groups (periods). There is a pattern that repeats for each group._

![International System of Numeration](international_system_of_numeration.png)

# Waypoint 2: Support Vietnamese Southern Numerals

The words "nghìn" and "linh" are the standard reading in Northern Vietnam, whilst "ngàn" and "lẻ" are more commonly used in the South.

Add an [optional argument](http://www.yourownlinux.com/2016/12/python-function-keyword-defaults-optional-arguments.html) `region` (a string) to the function `integer_to_vietnamese_numeral` that can be passed with one of the two possible values (strings):

- `'north'`: Northern Vietnam
- `'south'`: Southern Vietnam

By default, the parameter `region` is set to `north`. If the argument `region` is passed to the function `integer_to_vietnamese_numeral` with the value `None`, it defaults to `north`.

If the argument `region` passed to the function is not `None` and is not a string, the function `integer_to_vietnamese_numeral` raises an exception `TypeError`.

If the argument `region` passed to the function is not `None` and is not one of the two values `'north'` and `'south'`, the function `integer_to_vietnamese_numeral` raises an exception `ValueError`.

For examples:

```python
>>> integer_to_vietnamese_numeral(405)
'bốn trăm linh năm'
>>> integer_to_vietnamese_numeral(405, region='south')
'bốn trăm lẻ năm'
>>> integer_to_vietnamese_numeral(1971, region='north')
'một nghìn chín trăm bảy mươi một'
>>> integer_to_vietnamese_numeral(1971, region='south')
'một ngàn chín trăm bảy mươi một'
>>> integer_to_vietnamese_numeral(1, region=0)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "<input>", line 3, in _integer_to_vietnamese_numeral
TypeError: Argument "region" is not a string
>>> integer_to_vietnamese_numeral(1, region='whatever')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "<input>", line 6, in _integer_to_vietnamese_numeral
ValueError: Argument "region" has not a correct value
```

# Waypoint 3: Say Vietnamese Numeral

We would like our function `integer_to_vietnamese_numeral` to say the Vietnamese numeral that the function generates from the argument `n`. This is called [text-to-speech (TTS)](https://en.wikipedia.org/wiki/Speech_synthesis). Text-to-speech is [commonly used nowadays on mobile phones](https://www.youtube.com/watch?v=-gTn4Q-9Lk8).

Our implementation of the generation of synthesized speech from Vietnamese numerals will be simple. The function `integer_to_vietnamese_numeral` will successively play the sound of each Vietnamese numeral that the function has generated from the argument `n`.

The directory `./sounds/vie/north` contains several sound files corresponding to each Vietnamese numeral:

```bash
$ ls -la ./sounds/vie/north
total 152
drwxr-xr-x  21 dcaune  staff   672 Jun 24 10:25 .
drwxr-xr-x   5 dcaune  staff   160 May 27 08:52 ..
-rw-r--r--@  1 dcaune  staff  2496 Jun 24 08:52 ba.mp3
-rw-r--r--@  1 dcaune  staff  2784 Jun 24 08:54 bảy.mp3
-rw-r--r--@  1 dcaune  staff  2784 Jun 24 08:53 bốn.mp3
-rw-r--r--@  1 dcaune  staff  2880 Jun 24 08:55 chín.mp3
-rw-r--r--@  1 dcaune  staff  2880 Jun 24 08:52 hai.mp3
-rw-r--r--@  1 dcaune  staff  2688 Jun 24 10:24 không.mp3
-rw-r--r--@  1 dcaune  staff  2304 Jun 24 08:54 lăm.mp3
-rw-r--r--@  1 dcaune  staff  2112 Jun 24 08:52 mốt.mp3
-rw-r--r--@  1 dcaune  staff  2016 Jun 24 08:51 một.mp3
-rw-r--r--@  1 dcaune  staff  2976 Jun 24 08:56 mươi.mp3
-rw-r--r--@  1 dcaune  staff  2976 Jun 24 08:56 mười.mp3
-rw-r--r--@  1 dcaune  staff  2304 Jun 24 08:53 năm.mp3
-rw-r--r--@  1 dcaune  staff  2592 Jun 24 08:57 ngàn.mp3
-rw-r--r--@  1 dcaune  staff  2400 Jun 24 08:57 nghìn.mp3
-rw-r--r--@  1 dcaune  staff  3264 Jun 24 08:54 sáu.mp3
-rw-r--r--@  1 dcaune  staff  2592 Jun 24 08:55 tám.mp3
-rw-r--r--@  1 dcaune  staff  2496 Jun 24 08:57 trăm.mp3
-rw-r--r--@  1 dcaune  staff  2688 Jun 24 08:58 triệu.mp3
-rw-r--r--@  1 dcaune  staff  2208 Jun 24 08:58 tỷ.mp3
```

For example, the Vietnamese numeral for `128` is "một trăm hai mươi tám". If we pass the argument `n` with the value `128`, the function `integer_to_vietnamese_numeral` needs to play the following sound files in that order:

1. `một.mp3`
1. `trăm.mp3`
1. `hai.mp3`
1. `mươi.mp3`
1. `tám.mp3`

Update the function `integer_to_vietnamese_numeral` by adding an optional argument `activate_tts` (a boolean), with the default value `False`.

If the argument `activacte_tts` is passed with the value `None`, it defaults to `False`.

When this argument `activate_tts` is `True`, the function says the Vietnamese numeral that the function generates from the argument `n`, and it returns the Vietnamese numeral. If this argument `activate_tts` is `True`, the function only returns the Vietnamese numeral without saying it.

If the argument `activate_tts` passed to the function is not `None` and it is not a boolean value, the function `integer_to_vietnamese_numeral` raises an exception `TypeError`.

For example:

```python
>>> integer_to_vietnamese_numeral(405, activate_tts=True)
'bốn trăm linh năm'
>>> integer_to_vietnamese_numeral(405, region='south', activate_tts=True)
'bốn trăm lẻ năm'
>>> integer_to_vietnamese_numeral(1971, activate_tts=True, region='north')
'một nghìn chín trăm bảy mươi một'
>>> integer_to_vietnamese_numeral(1971, region='south', activate_tts=True)
'một ngàn chín trăm bảy mươi một'
>>> integer_to_vietnamese_numeral(1, activate_tts=1)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "<input>", line 3, in _integer_to_vietnamese_numeral
TypeError: Argument "activate_tts" is not a boolean
```

# Waypoint 4: Say Vietnamese Numeral with Southern Accent

The sound files contained in the directory `./sounds/vie` correspond to Vietnamese numerals said with the Northern accent. We would like to support Vietnamese Southern accent.

If the argument `region` is `'north'`, then the function `integer_to_vietnamese_numeral` says the numeral with the North accent, but if the argument `region` is `'south'`, then the function says the numeral with the South accent.

Unfortunately, we don't have sound files for Vietnamese Southern pronunciation of Vietnamese numerals. You need to create these audio files by recording yourself. You probably have an Android or a iPhone mobile phone. You will easily find a free voice recorder application on your favorite online store.

# Waypoint 5: Generate English Numerals

Write a function `integer_to_english_numeral` that takes an argument `n` representing a positive integer, and returns a string corresponding to the [English cardinal numeral](https://en.wikipedia.org/wiki/English_numerals) of this number.

The function raises the exception `TypeError` if the argument `n` is not an integer.

The function raises the exception `ValueError` if the argument `n` is not positive integer.

For example:

```python
>>> integer_to_english_numeral(96)
'ninety-six'
>>> integer_to_english_numeral(101)
'one hundred and one'
>>> integer_to_english_numeral(405)
'four hundred and five'
>>> integer_to_english_numeral(1971)
'one thousand and nine hundred and seventy-one'
>>> integer_to_english_numeral(5061)
'five thousand and sixty-one'
>>> integer_to_english_numberal('4096')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "<input>", line 3, in _integer_to_english_numeral
TypeError: Not an integer
>>> integer_to_english_numeral(-1)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "<input>", line 6, in _integer_to_english_numeral
ValueError: Not a positive integer
```

_Note: You can check whether your function `integer_to_english_numeral` returns the correct English cardinal numeral by comparing this result with [online generators](https://www.ego4u.com/en/cram-up/vocabulary/numbers/generator)._

# Waypoint 6: Say English Numerals

We would also want our function `integer_to_english_numeral` to say the English numeral that the function generates from the argument `n`! 😁

Update the function `integer_to_english_numeral` by adding an optional argument `activate_tts` (a boolean), with the default value `False`. If the argument`activate_tts`is passed with the value`None`, it defaults to`False`.

When this argument `activate_tts` is `True`, the function says the English numeral generated from the argument `n`, and the function returns the English numeral. If this argument `activate_tts` is `True`, the function only returns the English numeral without saying it.

If the argument `activate_tts` passed to the function is not `None` and it is not a boolean value, the function raises an exception `TypeError`.

Where are the audio files for the English cardinal numerals? Guess what, you will also have to record these audio files yourself and save them in the directory `./sounds/eng`... Like a pro! 😈😎


CHANGELOG.md
16/7/2019:
	Added function Voice North,South,Eng
	Added file sound Eng , South
	Bugs in playing sound Eng
	No documentation
17/7/2019:
	Added documentation
	Fixed bug Eng sound
	Merged WP1 to master


cardinal_numeral.py
import pygame

dict_of_value = {
    0: 'không',
    1: 'một',
    2: 'hai',
    3: 'ba',
    4: 'bốn',
    5: 'năm',
    6: 'sáu',
    7: 'bảy',
    8: 'tám',
    9: 'chín'
}
dict_of_north = {
    1: '',
    1000: 'nghìn',
    1000000: 'triệu',
    1000000000: 'tỷ'
}
dict_of_south = {
    1: '',
    1000: 'ngàn',
    1000000: 'triệu',
    1000000000: 'tỷ'
}
voice_of_north = {
    'không': './sounds/vie/north/khong.ogg',
    'linh': './sounds/vie/north/linh.ogg',
    'một': './sounds/vie/north/mot1.ogg',
    'mốt': './sounds/vie/north/mot2.ogg',
    'hai': './sounds/vie/north/hai.ogg',
    'ba': './sounds/vie/north/ba.ogg',
    'bốn': './sounds/vie/north/bon.ogg',
    'năm': './sounds/vie/north/nam.ogg',
    'lăm': './sounds/vie/north/lam.ogg',
    'sáu': './sounds/vie/north/sau.ogg',
    'bảy': './sounds/vie/north/bay.ogg',
    'tám': './sounds/vie/north/tam.ogg',
    'chín': './sounds/vie/north/chin.ogg',
    'trăm': './sounds/vie/north/tram.ogg',
    'nghìn': './sounds/vie/north/nghin.ogg',
    'triệu': './sounds/vie/north/trieu.ogg',
    'tỷ': './sounds/vie/north/ty.ogg',
    'mười':'./sounds/vie/north/muoi1.ogg',
    'mươi':'./sounds/vie/north/muoi2.ogg'
}
voice_of_south = {
    'không': './sounds/vie/south/khong.ogg',
    'lẻ': './sounds/vie/south/le.ogg',
    'một': './sounds/vie/south/mot1.ogg',
    'mốt': './sounds/vie/south/mot2.ogg',
    'hai': './sounds/vie/south/hai.ogg',
    'ba': './sounds/vie/south/ba.ogg',
    'bốn': './sounds/vie/south/bon.ogg',
    'năm': './sounds/vie/south/nam.ogg',
    'lăm': './sounds/vie/south/lam.ogg',
    'sáu': './sounds/vie/south/sau.ogg',
    'bảy': './sounds/vie/south/bay.ogg',
    'tám': './sounds/vie/south/tam.ogg',
    'chín': './sounds/vie/south/chin.ogg',
    'trăm': './sounds/vie/south/tram.ogg',
    'ngàn': './sounds/vie/south/ngan.ogg',
    'triệu': './sounds/vie/south/trieu.ogg',
    'tỷ': './sounds/vie/south/ty.ogg',
    'mười': './sounds/vie/south/muoi1.ogg',
    'mươi': './sounds/vie/south/muoi2.ogg'
}

def read_number_from_0_to_99(n):
    """
        input : number from 0 to 99

        output : string res pronnouce the number

    """

    res = []
    if n // 10 == 0:
        res.append(dict_of_value[n % 10])
    else:
        if n // 10 == 1:
            res.append('mười')
            if n % 10 == 5:
                res.append('lăm')
            elif n % 10 == 0:
                res.append('')
            else:
                res.append(dict_of_value[n % 10])
        else:
            res.append(dict_of_value[n // 10])
            res.append('mươi')
            if n % 10 == 5:
                res.append('lăm')
            elif n % 10 == 1:
                res.append('mốt')
            else:
                res.append(dict_of_value[n % 10])
    res = ' '.join(res)
    return res


def read_number_from_0_to_999(n, region):  # region : NORTH or SOUTH

    """
    :param n: number from 0 to 999
    :param region: if value = NORTH => pronounce by North Vietnamese, if value = SOUTH => pronounce by South Vietnamese
    :return: string res pronnouce the number

    """
    res = []
    res.append(dict_of_value[n // 100])
    res.append('trăm')
    temp = (n % 100) // 10
    if temp == 0:
        if n % 10 == 0:
            res.append('')
        elif region == 'NORTH':
            res.append('linh')
            res.append(read_number_from_0_to_99(n % 100))
        elif region == 'SOUTH':
            res.append('lẻ')
            res.append(read_number_from_0_to_99(n % 100))

    else:
        res.append(read_number_from_0_to_99(n % 100))
    res = ' '.join(res)
    return res


def integer_to_vietnamese_numeral(n, region, activate_tts):  # region : NORTH or SOUTH
    """

    :param n: number from 0 to 999,999,999,999
    :param region: if value = NORTH => pronounce by North Vietnamese, if value = SOUTH => pronounce by South Vietnamese
    :param activate_tts: if value = True => voice, if value = False => no voice
    :param activate_tts: if value is not boolean => raise TypeError
    :return: string res pronounce the number and voice

    """
    if isinstance(n, int) is False:
        raise KeyError('Not an integer')
    if n < 0:
        raise ValueError('Not a positive number')
    if not isinstance(activate_tts, bool):
        raise TypeError('Argument "activate_tts" is not a boolean')
    l = [1000000000, 1000000, 1000, 1]
    res = []
    i = 0
    while n // l[i] == 0:
        i += 1
    if n // l[i] >= 100:
        res.append(read_number_from_0_to_999(n // l[i], region))
    else:
        res.append(read_number_from_0_to_99(n // l[i]))
    if region == 'NORTH':
        res.append(dict_of_north[l[i]])
    else:
        res.append(dict_of_south[l[i]])
    n = n % l[i]
    i += 1
    while n != 0:
        if region == 'NORTH':
            if n//l[i]==0:
                i+=1
            else:
                res.append(read_number_from_0_to_999(n // l[i], region))
                res.append(dict_of_north[l[i]])
        if region == 'SOUTH':
            if n//l[i]==0:
                i+=1
            else:
                res.append(read_number_from_0_to_999(n // l[i], region))
                res.append(dict_of_south[l[i]])
        n = n % l[i]
        i += 1
    res = ' '.join(res)
    print(res)
    if activate_tts == True:
        pygame.init()
        l_voice = res.split(' ')
        l_voice.pop(-1)
        if region == 'NORTH':
            for i in l_voice:
                f_voice = voice_of_north[i]
                sound = pygame.mixer.Sound(f_voice)
                channel = sound.play()
                pygame.time.delay(500)
        if region == 'SOUTH':
            for i in l_voice:
                f_voice = voice_of_south[i]
                sound = pygame.mixer.Sound(f_voice)
                channel = sound.play()
                pygame.time.delay(800)
    return res


def integer_to_english_numeral(n):
    """

    :param n: number from 0 to 999,999,999,999
    :return: string res pronnouce the number

    """
    if not isinstance(n, int):
        raise TypeError("Not an integer")
    if n < 0:
        raise ValueError("Not a positive integer")
    else:
        dict_eng = {0: 'zero', 1: 'one', 2: 'two', 3: 'three', 4: 'four', 5: 'five',
             6: 'six', 7: 'seven', 8: 'eight', 9: 'nine', 10: 'ten',
             11: 'eleven', 12: 'twelve', 13: 'thirteen', 14: 'fourteen',
             15: 'fifteen', 16: 'sixteen', 17: 'seventeen', 18: 'eighteen',
             19: 'nineteen', 20: 'twenty',
             30: 'thirty', 40: 'forty', 50: 'fifty', 60: 'sixty',
             70: 'seventy', 80: 'eighty', 90: 'ninety'}
        n_thousand = 1000
        n_million = n_thousand * 1000
        n_billion = n_million * 1000
        n_trillion = n_billion * 1000

        if (n < 20):
            return dict_eng[n]

        if (n < 100):
            if n % 10 == 0:
                return dict_eng[n]
            else:
                return dict_eng[n // 10 * 10] + '-' + dict_eng[n % 10]

        if (n < n_thousand):
            if n % 100 == 0:
                return dict_eng[n // 100] + ' hundred'
            else:
                return dict_eng[n // 100] + ' hundred and ' + integer_to_english_numeral(n % 100)

        if (n < n_million):
            if n % n_thousand == 0:
                return integer_to_english_numeral(n // n_thousand) + ' thousand'
            else:
                return integer_to_english_numeral(n // n_thousand) + ' thousand and ' + integer_to_english_numeral(n % n_thousand)

        if (n < n_billion):
            if (n % n_million) == 0:
                return integer_to_english_numeral(n // n_million) + ' million'
            else:
                return integer_to_english_numeral(n // n_million) + ' million, ' + integer_to_english_numeral(n % n_million)

        if (n < n_trillion):
            if (n % n_billion) == 0:
                return integer_to_english_numeral(n // n_billion) + ' billion'
            else:

                return integer_to_english_numeral(n // n_billion) + ' billion, ' + integer_to_english_numeral(n % n_billion)
        if (n % n_trillion == 0):
            return integer_to_english_numeral(n // n_trillion) + ' trillion'
        else:
            return integer_to_english_numeral(n // n_trillion) + ' trillion, ' + integer_to_english_numeral(n % n_trillion)
dict_eng_voice = {
    'zero': './sounds/eng/zero.ogg',
    'one': './sounds/eng/one.ogg',
    'two': './sounds/eng/two.ogg',
    'three': './sounds/eng/three.ogg',
    'four': './sounds/eng/four.ogg',
    'five': './sounds/eng/five.ogg',
    'six': './sounds/eng/six.ogg',
    'seven': './sounds/eng/seven.ogg',
    'eight': './sounds/eng/eight.ogg',
    'nine': './sounds/eng/nine.ogg',
    'ten': './sounds/eng/ten.ogg',
    'eleven': './sounds/eng/eleven.ogg',
    'twelve': './sounds/eng/twelve.ogg',
    'thirteen': './sounds/eng/thirteen.ogg',
    'fifteen': './sounds/eng/fifteen.ogg',
    'sixteen': './sounds/eng/sixteen.ogg',
    'seventeen': './sounds/eng/seventeen.ogg',
    'eighteen': './sounds/eng/eighteen.ogg',
    'nineteen': './sounds/eng/nineteen.ogg',
    'twenty': './sounds/eng/twenty.ogg',
    'thirty': './sounds/eng/thirty.ogg',
    'fourty': './sounds/eng/fourty.ogg',
    'fifty': './sounds/eng/fifty.ogg',
    'sixty': './sounds/eng/sixty.ogg',
    'seventy': './sounds/eng/seventy.ogg',
    'eighty': './sounds/eng/eighty.ogg',
    'ninety': './sounds/eng/ninety.ogg',
    'hundred': './sounds/eng/hundred.ogg',
    'thousand': './sounds/eng/thousand.ogg',
    'million': './sounds/eng/million.ogg',
    'billion': './sounds/eng/billion.ogg',
    'trillion': './sounds/eng/trillion.ogg',
    'and': './sounds/eng/and.ogg'
}


def voice_of_integer_to_english_numeral(n, activate_tts=False):
    """
    :param n: number from 0 to 999,999,999,999
    :param activate_tts: if value = True => voice, if value = False => no voice
    :param activate_tts: if value is not boolean => raise TypeError
    :return: string res pronounce the number and voice

    """
    res = integer_to_english_numeral(n)
    print(res)
    if not isinstance(activate_tts, bool):
        raise TypeError('Argument "activate_tts" is not a boolean')
    elif activate_tts is True:
        l_voice = res.replace('-', ' ').replace(',', '').split(' ')
        pygame.init()
        for i in l_voice:
            sound = pygame.mixer.Sound(dict_eng_voice[i])
            channel = sound.play()
            clock = pygame.time.Clock().tick(1.234567890)
            channel = sound.stop()
        pygame.quit()
    else:
        return res

# integer_to_vietnamese_numeral(102015, 'NORTH', False)
# integer_to_vietnamese_numeral(102015, 'SOUTH', False)
# integer_to_vietnamese_numeral(102015, 'NORTH', True)
# integer_to_vietnamese_numeral(102015, 'SOUTH', True)
# voice_of_integer_to_english_numeral(102015,False)
# voice_of_integer_to_english_numeral(102015,True)
# integer_to_vietnamese_numeral(-102015, 'NORTH', False)
# integer_to_vietnamese_numeral('102015', 'NORTH', False)
# integer_to_vietnamese_numeral(102015, 'NORTH', 122121)
